cardano-db/src/Cardano/Db/Migration/Extra/CosnumedTxOut/Queries.hs

```diff
@@ -202,40 +206,66 @@ countConsumed = do
     pure countRows
   pure $ maybe 0 unValue (listToMaybe res)
 
+deleteAndUpdateConsumedTxOut :: MonadIO m => Trace IO Text -> Bool -> Word64 -> ReaderT SqlBackend m ()
+deleteAndUpdateConsumedTxOut trce hasConsumedField blockNoDiff = do
+  maxTxInId <- findMaxTxInId blockNoDiff
+  case maxTxInId of
+    Left errMsg -> liftIO $ logInfo trce $ "No tx_out was deleted: " <> errMsg
+    Right mxtxid -> do
+      migrateNextPage mxtxid 0
+  where
+    migrateNextPage :: MonadIO m => TxInId -> Word64 -> ReaderT SqlBackend m ()
+    migrateNextPage mxTxInId offst = do
+      page <- getInputPage offst pageSize
+      mapM_ (handlePageEntry mxTxInId) page
+      when (fromIntegral (length page) == pageSize) $
+        migrateNextPage mxTxInId $!
+          offst
+            + pageSize
+
+    handlePageEntry :: MonadIO m => TxInId -> (TxInId, TxId, Word64) -> ReaderT SqlBackend m ()
+    handlePageEntry maxTxInId (txInId, txId, index)
+      | txInId <= maxTxInId = deleteTxOutConsumed txId index
+      | otherwise = do
+          when hasConsumedField createConsumedTxOut
```

Comment:
I'm a bit confused. `createConsumedTxOut` should be called if the field doesn't exist. Also `handlePageEntry` being a iterative function can still call `createConsumedTxOut` more than once.

Reply:
I was looking for a way to know if it should `createConsumedTxOut` from `syncEnv` then it would never create more than once.